src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Samples.cshtml.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using ApiView;
+using APIView;
+using APIViewWeb.Models;
+using APIViewWeb.Repositories;
+using Markdig.Helpers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json.Linq;
+
+namespace APIViewWeb.Pages.Assemblies
+{
+    public class UsageSamplePageModel : PageModel
+    {
+        private readonly UsageSampleManager _samplesManager;
+        private readonly ReviewManager _reviewManager;
+        private const string ENDPOINT_SETTING = "Endpoint";
+        private readonly CommentsManager _commentsManager;
+        private readonly NotificationManager _notificationManager;
+        public readonly UserPreferenceCache _preferenceCache;
+
+        public string Endpoint { get; }
+        public ReviewModel Review { get; private set; }
+        public UsageSampleModel Sample { get; private set; }
+        public IEnumerable<UsageSampleModel> SampleRevisions { get; private set; }
```

Comment:
Please see my comment above on UsageSampleModel for a suggestion on how to maintain revisions in the same way we maintain review revisions. Adding revisions in page model tightly couple that to this specific page model and it will make it difficult for reuse in different page in the future. Having revisions as a property within sample model makes it more easy for use in different page or even on the server side.